Automattic/jetpack
# Site Editor: do not display Jetpack sidebar panels when not needed

## Proposed changes:

Following recent changes in the Gutenberg plugin, we currently display some Jetpack sidebar panels in the site editor, when they do not make sense because they would only apply to specific posts or pages:

- The QR code panel
- The Social Previews panel
- The SEO Tools panel
- The AI Assistant panel.

While Gutenberg will address that issue in https://github.com/WordPress/gutenberg/pull/62531, it would be nice for us to add a safeguard, as recommended here:
https://make.wordpress.org/core/2024/06/18/editor-unified-extensibility-apis-in-6-6/

### Other information:


## Jetpack product discussion

* Epic: #36721
* Related issue: #37713

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

You can test this on a site running the Gutenberg plugin version 18.5.

* Go to Appearance > Editor
* Check if there are any Jetpack sidebar elements.
    * You should see none with this PR applied.
